proto/ord_schema.proto

```diff
@@ -60,50 +82,148 @@ message ReactionInput {
-  // Any accepted common name, trade name, etc.
-  string name = 5;
-  string cas_number = 6;
-  int32 pubchem_sid = 7;
-  int32 chemspider_id = 8;
-  message Amount {
-    // TODO(kearnes): Should this be a float value and a type enum?
-    oneof value {
-      float mass_in_g = 1;
-      float moles = 2;
-      float volume_in_ml = 3;
-      float concentration_in_m = 4;
-    }
-    Compound solvent = 5;
-    // For catalysts on supports.
-    Compound support = 6;
-    float weight_percent = 7;
-    // Precision of the measurement, as a percentage of the value.
-    float precision_in_percent = 8;
-  }
-  Amount amount = 9;
-  enum CompoundPreparation {
+  repeated CompoundIdentifier identifiers = 1;
+  Amount amount = 2;
+  /**
+   * Compounds may undergo additional preparation before being used in a reaction
+   * after being received from a supplier or vendor. 
+   */
+  enum Preparation {
     COMPOUND_PREPARATION_UNKNOWN = 0;
-    COMPOUND_PREPARATION_NONE = 1;
+    COMPOUND_PREPARATION_NONE = 1;  // Compound used as received.
     COMPOUND_PREPARATION_REPURIFIED = 2;
     COMPOUND_PREPARATION_SPARGED = 3;
-    COMPOUND_PREPARATION_DRIED = 4;
+    COMPOUND_PREPARATION_DRIED = 4;  // Moisture removed, e.g., using molecular sieves.
   }
-  CompoundPreparation preparation = 10;
-  string vendor_source = 11;
-  string vendor_lot = 12;
-  message CompoundFeature {
+  Preparation preparation = 3;
+  string preparation_custom = 4;
+  string vendor_source = 5;
+  string vendor_lot = 6;
+  /**
+   * Compounds can accommodate any number of features. These may include simple
+   * properties of the compound (e.g., molecular weight), heuristic estimates
+   * of physical properties (e.g., ClogP), optimized geometries (e.g., through
+   * DFT), and calculated stereoselectronic descriptors.
+   */
+  message Feature {
     string name = 1;
     oneof kind {
       string string_value = 2;
       float float_value = 3;
     }
     string how_computed = 4;
   }
-  repeated CompoundFeature features = 13;
+  repeated Feature features = 7;
+}
+
+/**
+ * Compound identifiers uniquely define a single (pure) chemical species.
+ * While we encourage the use of SMILES strings, these do not work well in
+ * all cases (e.g., handling tautomerism, axial chirality). Multiple
+ * identifiers may be specified for a single compound to avoid ambiguity.
+ * We discourage chemicals from being defined only by a name.
+ */
+message CompoundIdentifier {
+  // Possible identifier types are listed in an enum for extensibility
+  enum IdentifierType {
+    COMPOUND_IDENTIFIER_TYPE_UNKNOWN = 0;  // Any other identifier
+    COMPOUND_IDENTIFIER_TYPE_SMILES = 1;  //  Simplified molecular-input line-entry system.
+    COMPOUND_IDENTIFIER_TYPE_INCHI = 2;  // IUPAC International Chemical Identifier.
+    COMPOUND_IDENTIFIER_TYPE_MOLFILE = 3;  // MDL Molfile V3000.
+    COMPOUND_IDENTIFIER_TYPE_IUPAC_NAME = 4;  // Chemical name following IUPAC nomenclature recommendations.
```

Comment:
Let's keep an 80 char limit; in this case we can move the comments to their own line above (and maybe be consistent within the enum)